Indicators/CommodityChannelIndex.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+
+
+namespace QuantConnect.Indicators {
+    /// <summary>
+    ///     Represents the traditional commodity channel index (CCI)
+    ///     
+    ///     CCI = (Typical Price - 20-period SMA of TP) / (.015 * Mean Deviation)
+    ///     Typical Price (TP) = (High + Low + Close)/3
+    ///     Constant = 0.015
+    ///
+    ///     There are four steps to calculating the Mean Deviation, first, subtract
+    ///     the most recent 20-period average of the typical price from each period's
+    ///     typical price. Second, take the absolute values of these numbers. Third,
+    ///     sum the absolute values. Fourth, divide by the total number of periods (20).
+    /// </summary>
+    public class CommodityChannelIndex : TradeBarIndicator {
+
+        /// <summary>This constanst is used to ensure that CCI values fall between +100 and -100, 70% to 80% of the time</summary>
+        private readonly decimal _k = 0.015m;
+        private readonly int _period;
+
+        /// <summary>
+        /// Gets the type of moving average
+        /// </summary>
+        public MovingAverageType MovingAverageType { get; private set; }
+
+
+        /// <summary>
+        /// Keep track of the simple moving average of the typical price
+        /// </summary>
+        private readonly IndicatorBase<IndicatorDataPoint> _typicalPriceAverage;
+
+        /// <summary>
+        /// Keep track of the mean absolute deviation of the typical price
+        /// </summary>
+        private readonly IndicatorBase<IndicatorDataPoint> _typicalPriceMeanDeviation;
```

Comment:
Yeah that makes sense.  I got the example looking at the ATR implementation which has the private _smoother.  Should that have the same private/public properties?  In general what's the deciding factor to make them private or public?